runtime/repl.go

```diff
@@ -66,15 +67,18 @@ func NewREPL(
 
 	storage := interpreter.NewInMemoryStorage(nil)
 
-	// NOTE: storage option must be provided *before* the predeclared values option,
-	// as predeclared values may rely on storage
+	// necessary now due to log being looked up in the
+	// interpreter's activations instead of the checker
+	baseActivation := interpreter.NewBaseActivation()
```

Comment:
Instead of recreating a whole new base activation, it's cheaper to just create a new activation and set the base activation as the parent:

```suggestion
	baseActivation := activations.NewActivation[*interpreter.Variable](nil, interpreter.BaseActivation)
```